tests/backup/backup_resiliency_test.go

```diff
@@ -1298,3 +1281,263 @@ var _ = Describe("{RebootNodesWhenBackupsAreInProgress}", func() {
+			err = CreateSourceAndDestClusters(orgID, "", "", ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Creating source cluster %s and destination cluster %s", SourceClusterName, destinationClusterName))
+			srcClusterStatus, srcClusterUid = Inst().Backup.RegisterBackupCluster(orgID, SourceClusterName, "")
+			dash.VerifyFatal(srcClusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying source cluster %s status", SourceClusterName))
+			destClusterStatus, _ = Inst().Backup.RegisterBackupCluster(orgID, destinationClusterName, "")
+			dash.VerifyFatal(destClusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying destination cluster %s status", destinationClusterName))
+		})
+		Step("Getting the replica factor of px-backup deployment in backup namespace before taking backup", func() {
+			pxBackupNS, err = backup.GetPxBackupNamespace()
+			log.FailOnError(err, "Getting backup namespace")
+			log.Infof("Getting the replica factor of px-backup deployment in backup namespace [%s] before taking backup", pxBackupNS)
+			backupDeployment, err = apps.Instance().GetDeployment(pxBackupDeployment, pxBackupNS)
+			originalReplicaCount = *backupDeployment.Spec.Replicas
+			log.Infof("Replica count for px-backup pod before taking backup is %v", originalReplicaCount)
+			log.FailOnError(err, fmt.Sprintf("Getting px-backup deployment replica in backup namespace %s", pxBackupNS))
+			dash.VerifyFatal(originalReplicaCount > scaledDownReplica, true, "Verifying px_backup deployment replica before taking backup")
+		})
+		Step("Taking backup of applications", func() {
+			log.InfoD("Taking backup of applications")
+			ctx, err = backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, namespace := range appNamespaces {
+				for i := 0; i < numberOfBackups; i++ {
+					sem <- struct{}{}
+					backupName := fmt.Sprintf("%s-%s-%d-%v", BackupNamePrefix, namespace, i, time.Now().Unix())
+					backupNames = append(backupNames, backupName)
+					wg.Add(1)
+					go func(backupName string, namespace string) {
+						defer GinkgoRecover()
+						defer wg.Done()
+						defer func() { <-sem }()
+						_, err = CreateBackupWithoutCheck(backupName, SourceClusterName, bkpLocationName, backupLocationUID,
+							[]string{namespace}, labelSelectors, orgID, srcClusterUid, "", "", "", "", ctx)
+						dash.VerifyFatal(err, nil, fmt.Sprintf("Taking backup %s of application- %s", backupName, namespace))
+					}(backupName, namespace)
+				}
+			}
+			wg.Wait()
+			log.InfoD("The list of backups taken are: %v", backupNames)
+		})
+		Step("Scaling px-backup deployment replica count to 0 and back to original replica while backup is in progress", func() {
+			log.InfoD("Scaling px-backup deployment replica count to 0 while backup is in progress")
+			*backupDeployment.Spec.Replicas = scaledDownReplica
+			backupDeployment, err = apps.Instance().UpdateDeployment(backupDeployment)
```

Comment:
How about renaming the variable backupDeployment here to updatedBackupDeployment?

Reply:
Done.